apps/labs/posts/sparse-array-ecosystem.md

```diff
@@ -0,0 +1,231 @@
+
+However, the format has some disadvantages:
+
+1. Random access using a coordinate isn't possible. One has to search for an existing coordinate. This may be easy if the coordinates are sorted and hard if they're not.
+2. Random insertion causes the coordinates to be inserted out of order.
+3. Sorting may be frequently required.
+
+### Other Formats
+
+There are many other formats, each with their own set of advantages and disadvantages:
+
+- CSR/CSC are popular 2-dimensional formats, with fast row-major or column-major access respectively.
+- Blocked formats are versions of formats which can store dense regions in a blocked structure.
+
+It is, however, important to recognize that we aren't limited to the formats we mention here. You can come up with your own way of encoding the positions and thus your own special brand of sparse array format!
+
+## Python Libraries for Sparse Arrays
+
+While choosing the right format can be tricky, thankfully, Python has a number of libraries providing sparse array storage and compute capabilities. Some of these libraries have been around for some time, while others are relatively new to the scene. I'll briefly discuss a few of the major players.
+
+### [`sparse`](https://sparse.pydata.org/)
+
+`sparse` (also known as PyData/Sparse) is a library maintained by yours truly. Its aim is to provide perfomant sparse array operations following the Array API standard, though we are some time away from reaching that goal. The work on this library started in April 2017, with the author of this blog post taking over maintenance around January 2018. `sparse` provides a drop-in replacement for NumPy arrays, with support for **n-dimensional** arrays.
+
+`sparse` provides a few main options for constructing arrays:
+
+```python
+# 1. with `sparse.asarray`
+spx = sparse.asarray(x, [format=...])
+
+# 2. Create a `sparse.DOK` and then convert
+spy = sparse.DOK(shape=..., dtype=...)  # Instantiate
+spy[...] = ...  # Assign repeatedly
+spy = spy.asformat(...)  # Convert
+
+# 3. By specifying the constituent arrays
+spz = sparse.COO(shape=..., data=..., coords=...)
+```
+
+One current pain point for `sparse` is its performance: it's notably slower than equivalent SciPy operations. Work is ongoing to resolve this difference in performance while maintaining generality.
+
+### [`scipy.sparse`](https://docs.scipy.org/doc/scipy/reference/sparse.html)
+
+`scipy.sparse` is one of the original sparse matrix libraries. The library has existed since before SciPy was moved to GitHub, or even TRAC, with the earliest references pointing to 2006. It is built on NumPy matrices and has been limited to two dimensional sparse arrays. Work is underway to migrate `scipy.sparse` away from NumPy matrices (a discouraged corner of NumPy's API) and to conform to a more modern API. Details can be found here in [various discussion posts on the SciPy Discourse](https://discuss.scientific-python.org/tag/sparse-arrays).
```

Comment:
```suggestion
### scipy.sparse

[`scipy.sparse`](https://docs.scipy.org/doc/scipy/reference/sparse.html) is one of the original sparse matrix libraries. The library has existed since before SciPy was moved to GitHub, or even TRAC, with the earliest references pointing to 2006. It is built on NumPy matrices and has been limited to two dimensional sparse arrays. Work is underway to migrate `scipy.sparse` away from NumPy matrices (a discouraged corner of NumPy's API) and to conform to a more modern API. Details can be found here in [various discussion posts on the SciPy Discourse](https://discuss.scientific-python.org/tag/sparse-arrays).
```